blog/2024-01-24-transforming-your-data-1/index.md

```diff
@@ -0,0 +1,437 @@
+---
+slug: transforming-your-data-to-graphs-1
+authors: 
+  - prashanth
+tags: [use-case]
+---
+
+import RelationalSchema from './transforming_data_relational_schema.png';
+import GraphSchema from './transforming_data_graph_schema.png';
+import EdgeTables from './transforming_data_edge_tables.png';
+import GraphSchemaViz from './transforming_data_schema_viz.png';
+import GraphDataViz from './transforming_data_graph_viz.png';
+
+# Transforming your data to graphs: ETL
+
+Ever since the birth of database management systems (DBMSs), tabular relations and graphs have been
+the core data structures used to model application data in two broad classes of systems:
+Relational (RDBMS) and Graph (GDBMS).
+
+In this post, we'll look at how to transform data that might exist in a typical relational system
+to a graph and load it into a Kùzu database. The aim of this post and the next one is to showcase
+"graph thinking"[^1], where you explore existing connections in your data, and how it can potentially
+help uncover new insights.
+
+<!-- truncate -->
+
+## When are graphs useful?
+
+A lot of existing enterprise data is in the form of relational tables. In RDBMSs, connections
+between entities are often implicitly defined by the schema of the tables via foreign key
+constraints.
+
+In many cases, graphs are the most natural data structures to represent structured data by defining
+*explicit* relationships between entities. While the relational data model requires strict
+schematization of the data, graph models provide more flexibility, especially when storing sparse
```

Comment:
I edited this section as per your feedback. The message appeared a bit disjointed based on the prior paragraph, so I organized it a bit more (keeping all parts) so that we more clearly describe the benefits of a graph data model.